apps/web/modules/ee/contacts/api/client/[environmentId]/user/lib/update-user.ts

```diff
@@ -0,0 +1,135 @@
+import { contactCache } from "@/lib/cache/contact";
+import { updateAttributes } from "@/modules/ee/contacts/lib/attributes";
+import { prisma } from "@formbricks/database";
+import { getEnvironment } from "@formbricks/lib/environment/service";
+import { getOrganizationByEnvironmentId } from "@formbricks/lib/organization/service";
+import { ResourceNotFoundError } from "@formbricks/types/errors";
+import { getContactByUserIdWithAttributes } from "./contact";
+import { getUserState } from "./user-state";
+
+export const updateUser = async (
+  environmentId: string,
+  userId: string,
+  device: "phone" | "desktop",
+  attributes?: Record<string, string>
+) => {
+  const environment = await getEnvironment(environmentId);
+
+  if (!environment) {
+    throw new ResourceNotFoundError(`environment`, environmentId);
+  }
+
+  const organization = await getOrganizationByEnvironmentId(environmentId);
+
+  if (!organization) {
+    throw new ResourceNotFoundError(`organization`, environmentId);
+  }
+
+  let contact = await getContactByUserIdWithAttributes(environmentId, userId);
+
+  if (!contact) {
+    contact = await prisma.contact.create({
+      data: {
+        environment: {
+          connect: {
+            id: environmentId,
+          },
+        },
+        attributes: {
+          create: [
+            {
+              attributeKey: {
+                connect: { key_environmentId: { key: "userId", environmentId } },
+              },
+              value: userId,
```

Comment:
I think we should also add the attributes here we already know about and that are passed to the endpoint. This would reduce the number of database calls if we need to update the attributes afterwards.